### DISCUSSION TITLE
MySQL connector data type mapping questions

### DISCUSSION THREAD

**Discussion by langbai20** [trinodb/trino#17893] [Q&A]
Trino Version: 403

The problem is described as follows:
We are migrating presto cluster tasks into trino cluster, but we met some questions of differences between the two computing engines. 
First of all, Trino Mysql connector type mapping BIGINT UNSIGNED to DECIMAL(20, 0), but presto is still BIGINT.

We have a large number of python(pandas) query tasks by trino engine, pandas is very strict about data type handling, so we expect to find a config parameter that maps DECIMAL(20, 0) to BIGINT (or other integer data), to avoid having to modify a lot of historical code.

At the same time, we also expect to find a config parameter that maps varchar(1) to unbounded varchar, I only find parameter "jdbc-types-mapped-to-varchar", if we configure "jdbc-types-mapped-to-varchar=varchar", it will bring reduction of querying speed, Is there any parameter to mapping varchar(x)? 

<img width="678" alt="image" src="https://github.com/trinodb/trino/assets/88966261/d07ecf3d-8212-4496-865c-36d96b481089">

Look forward to the community help, thanks!

**Comment by ebyhr** [comment]
> we expect to find a config parameter that maps DECIMAL(20, 0) to BIGINT (or other integer data)

There's no such a config property and also no plan to support it. 

> we also expect to find a config parameter that maps varchar(1) to unbounded varchar,

This is also not supported in the connector because MySQL JDBC driver returns `VARCHAR` in `DATA_TYPE` regardless of the length.
